### PR TITLE
update sdk version and refine the example code

### PR DESCRIPTION
**Pull request by countcain**

### CODE REVIEW

File: App.js [hunk 1]
```diff
@@ -38,7 +42,7 @@ export default class App extends Component {
     this.sdkCrashEventSubscription.remove();
```
Comment (sebouh00) [inline App.js:42]: @countcain When `setupSdk` does not complete and we exit `componentDidMount` via the `catch` block, this line here, along with the preceding two, will result in an error due to these variables not having been initialized (`subscribeAfterSDKSetup` will never have been called).

File: App.js [hunk 2]
```diff
@@ -143,10 +122,10 @@ export default class App extends Component {
         PermissionsAndroid.PERMISSIONS.ACCESS_FINE_LOCATION
       );
       if (granted === PermissionsAndroid.RESULTS.GRANTED) {
-        const sdkInitialized = await this.isSdkInitialized();
-        if (sdkInitialized) {
+        const sdkNotInitialized = await this.isSdkNotInitialized();
+        if (!sdkNotInitialized) {
```
Comment (sebouh00) [inline App.js:126]: The double negative is confusing here. Additionally, this condition shouldn't pass if the SDK is still initializing. I suggest adding a second method called `isSdkInitialized` that explicitly checks for the `INITIALIZED` state.

File: App.js (hunk 1 above)
Comment (countcain) [reply App.js:45]: Yes. the app will crash here.

**Comment by countcain** [comment]
@sebouh00 `isSdkInitialized` and the subscription checks have been added.

**Comment by sebouh00** [comment]
@countcain These changes will be merged to `user-linking` too, right?

**Comment by countcain** [comment]
> @countcain These changes will be merged to `user-linking` too, right?

Nope. This one goes to master. `user-linking` was branched out from this branch. So every change here will also be included.

**Pull request merged**